### ISSUE TITLE
Early stopping

### ISSUE THREAD

**Issue by nathanielsimard** [burn-rs/burn#841]
Add the option to keep the best weighs ranked by validation loss.
Stop the training when the X following epochs don't improve the metric.

## Optional

Being able to set the metric to be used for ranking.
